Add `svg` component reference (#9911)
withastro/docs: src/content/docs/en/guides/images.mdx

<ReadMore>See details about [the `<Picture />` component properties](/en/reference/modules/astro-assets/#picture-properties) in the `astro:assets` reference.</ReadMore>

> ### Inline `.svg` files in the HTML

Reviewer: ```suggestion
### Inline `.svg` files in the HTML (experimental)
```
Looking at the source PR, it seems this will be behind an experimental flag?

CURRENTLY, **everything** about experimental features is only documented in config reference docs, but I know everyone hates that. So for v5 we can instead do something like this, but then we'll also need to link back to experimental config reference docs somehow to tell people how they can enable it.

OR **all** of this documentation exists only within the JSDOC of that PR.

<p><Since v="5.0.0" /></p>

> To inline local `.svg` files directly in your HTML, SVG files can be imported and used like any other [Astro components](/en/basics/astro-components/#component-based-design).

Reviewer: ```suggestion
To inline local `.svg` files directly in your HTML, import the SVG file to use as an Astro component. This process assigns an `<svg>` HTML tag to a component, and requires the same conventions (e.g. capitalization) as when [using dynamic tags](/en/basics/astro-syntax/#dynamic-tags).
```
I'm assuming this is how this works under the hood, and I think the link to the explanation and caveats there is more helpful, if true!

```astro
---
import Logo from '../assets/logo.svg';
---

> <Logo />

Reviewer: Question: what happens if no attributes are passed? Is meta data from the `.svg` file used so that the image is basically presented as is? Are there any required attributes, or will this just work like this?

I'm not sure it's entirely necessary to actually state more here as long as someone can literally just do this and get SOME kind of image (even if not the way they want it to eventually look). Just want to make sure nothing is required, and everything else is an optional tweak of the logo file.

Reviewer 2: If nothing is passed it will just use any attributes that are on the SVG file. I think this is the expected functionality.

```

You can set attributes such as `width`, `height`, `fill`, `stroke`, and other common SVG attributes on the imported component. These attributes will automatically be applied to the underlying `<svg>` element. Properties passed to the component will override duplicate properties that exist in the original `.svg` file.

```astro
<Logo width={64} height={64} fill="currentColor" />
```

> As a convenience, SVG components accept a `size` property. `size` is a shorthand which sets both the `width` and `height` properties to the same value.

Reviewer: ```suggestion
As a convenience, SVG components also accept a `size` property. `size` is a shorthand which sets both the `width` and `height` properties to the same value.
```
Such a tiny change, but feels more natural to me!

> **Example:**

Reviewer: ```suggestion
The following example uses Astro's `size` property to set an equal width and height of 48 instead of setting the HTML `<svg>` attributes of `width` and `height` separately:
```

## Community Integrations

There are several third-party [community image integrations](https://astro.build/integrations?search=images) for optimizing and working with images in your Astro project.

> ## Inline `.svg` files in the HTML (experimental)

Author: ```suggestion
## Using `.svg` files as components (experimental)
```

Is this a more accurate title?

Reviewer: Ooh, much better. (Though, on its own page, we won't need a heading at all. But rather a page title and description).

I'm just about to commit the standalone Experimental Flag PRs, so you'll see what we need.

This feature allows you to import SVG files directly into your Astro project. By default, Astro will inline the SVG content into your HTML output.

To enable this feature, set `experimental.svg` to `true` in your Astro config:

 ```js
{
  experimental: {
    svg: true,
  },
}
```

> To use this feature, import an SVG file to use as an Astro component, passing any common SVG attributes. Astro also provides a `size` attribute to set equal `height` and `width` properties. This process assigns an `<svg>` HTML tag to a component, and requires the same conventions (e.g. capitalization) as when [using dynamic tags](/en/basics/astro-syntax/#dynamic-tags).

Author: ```suggestion
To use this feature, reference the default import of any local `.svg` file. Since this import is treated as an Astro component, it requires using the same conventions (e.g. capitalization) as when [using dynamic tags](/en/basics/astro-syntax/#dynamic-tags).
```

Simplified for clarity and avoids repetition.

Reviewer: Love it!

```astro
---
import Logo from './path/to/svg/file.svg';
---

> <Logo size={24} />

Author: ```suggestion
<Logo />
```

No props?

```

### SVG component attributes

> You can set attributes such as `width`, `height`, `fill`, `stroke`, and other common SVG attributes on the imported component. These attributes will automatically be applied to the underlying `<svg>` element. Properties passed to the component will override duplicate properties that exist in the original `.svg` file.

Author: ```suggestion
You can pass props such as `width`, `height`, `fill`, `stroke`, and any other attribute accepted by the [native `<svg>` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg). These attributes will automatically be applied to the underlying `<svg>` element. If a property is present in the original `.svg` file and is passed to the component, the value passed to the component will override the original value.
```

Clarity

Reviewer: Crystal clear.
